source/runtime.hpp

```diff
@@ -31,14 +31,70 @@ namespace reshade
 	class runtime abstract
 	{
 	public:
+		enum class game
+		{
+			ACO,                // Assassin's Creed Origins
+			PCARS2,				// Project Cars 2
+			PCARS2AVX,			// Project Cars 2 - other exe
+			TITANFALL2,			// TitanFall 2
+			TITANFALL2_TRIAL,   // TitanFall 2 - trial exe
+			SOW,     			// Shadow Of War
+			ELEX,     			// Elex
+			AOM_RELEASE_FINAL,  // Agents Of Mahyem
+			AOM,     			// Agents Of Mahyem - other exe
+			UNKNOWN
+		};
+
+		enum depth_buffer_retrieval_mode {
+			POST_PROCESS = 0,
+			BEFORE_CLEARING_STAGE = 1,
+			AT_OM_STAGE = 2
+		};
+
+		enum depth_buffer_texture_type {
+			BOTH = 0,
+			DEPTH_BUFFER = 1,
+			STENCIL_BUFFER = 2
+		};
+
+		enum class allowed_technique
+		{
+			DISPLAY_DEPTH,
+			SUPER_DEPTH_3D,
+			POLYNOMIAL_BARREL_DISTORTION_M,
+			POLYNOMIAL_BARREL_DISTORTION_S,
+			MXAO
+		};
+
 		/// <summary>
 		/// File path to the current module.
 		/// </summary>
 		static filesystem::path s_reshade_dll_path;
 		/// <summary>
+		/// screen width
+		/// </summary>
+		static unsigned int screen_width;
+		/// <summary>
+		/// screen height
+		/// </summary>
+		static unsigned int screen_height;
+		/// <summary>
 		/// File path to the current executable.
 		/// </summary>
 		static filesystem::path s_target_executable_path;
+		/// <summary>
+		/// Depth buffer retrieval mode (post process, before depth Buffer Clearing)
+		/// </summary>
+		static unsigned int depth_buffer_retrieval_mode;
+		/// <summary>
+		/// Depth buffer clearing number
+		/// </summary>
+		static unsigned int depth_buffer_clearing_number;
+		/// <summary>
+		/// Depth buffer texture format
+		/// </summary>
+		static unsigned int depth_buffer_texture_format;
```

Comment:
Why are all these `static`. Please make them normal fields in `runtime`. They may even be much better suited as fields in `d3d11_runtime` since they are currently specific to that implementation!! This would make it hard to get access to them in ImGUI though.

Reply:
Yes, you are totally right. Setting these to static is useless...

Reply:
For the two other params, i need them in different part of Reshade, so that's why i prefered to set them static.